paasta_tools/marathon_tools.py

```diff
@@ -766,17 +776,42 @@ def deformat_job_id(job_id):
     return decompose_job_id(job_id)
 
 
-def read_namespace_for_service_instance(name, instance, cluster=None, soa_dir=DEFAULT_SOA_DIR):
-    """Retreive a service instance's nerve namespace from its configuration file.
-    If one is not defined in the config file, returns instance instead."""
+def read_all_namespaces_for_service_instance(name, instance, cluster=None, soa_dir=DEFAULT_SOA_DIR):
```

Comment:
This function is interesting. I wonder why we re-parsed the marathon.yaml instead of loading a MarathonServiceConfig object.

Seems like this could just be:

```
def read_all_namespaces_for_service_instance(name, instance, cluster=None, soa_dir=DEFAULT_SOA_DIR):
    if not cluster:
          cluster = load_system_paasta_config().get_cluster()
    marathon_service_config = load_marathon_service_config(service, instance, cluster, load_deployments=False, soa_dir=soa_dir)
    return marathon_service_config.get_registration_namespaces()
```